get_started_with_membrane/04_bins.md

```diff
@@ -0,0 +1,19 @@
+# Bins
+
+Bins, similarly to pipelines, are containers for elements. However, at the same time, they can be placed and linked within pipelines. Although a bin is a separate Membrane entity, it can be perceived as a pipeline within an element. Bins can also be nested within one another, though we don't recommend too much nesting, as it may end up hard to maintain. For an example of a bin, have a look at the [RTMP source bin](https://github.com/membraneframework/membrane_rtmp_plugin/blob/master/lib/membrane_rtmp_plugin/rtmp/source/bin.ex) or [HTTP adaptive stream sink bin](https://github.com/membraneframework/membrane_http_adaptive_stream_plugin/blob/master/lib/membrane_http_adaptive_stream/sink_bin.ex).
+
+The main use cases for a bin are:
+- creating reusable element groups,
+- encapsulating children's management logic, for instance, dynamically spawning or replacing elements as the stream changes.
+
+## Bin's pads
+
+Bin's pads are defined and linked similarly to element's pads. However, their role is limited to proxy the stream to other elements and bins inside (inputs) or outside (outputs). To achieve that, each input pad of a bin needs to be linked to both an output pad from the outside of a bin and an input pad of its child inside. Accordingly, each bin's output should be linked to output inside and input outside of the bin.
```

Comment:
It would be nice to have an example showing how to link bin's pad from inside, using `bin_input/1` and `bin_output/2`

Reply:
I'll do that in a separate PR